picard/acoustid/manager.py

```diff
@@ -82,16 +84,32 @@ def submit(self):
         if not submissions:
             self._check_unsubmitted()
             return
-        log.debug("AcoustID: submitting ...")
+        log.debug("AcoustID: submitting total of %d fingerprints...", len(submissions))
+        self._batch_submit(submissions)
+
+    def _batch_submit(self, submissions):
+        if not submissions:  # All fingerprints submitted, nothing to do
+            log.debug("AcoustID: submitted all fingerprints")
+            self.tagger.window.set_statusbar_message(
+                N_('AcoustIDs successfully submitted.'),
+                echo=None,
+                timeout=3000
+            )
+            self._check_unsubmitted()
+            return
+        submission_batch = submissions[:self.BATCH_SUBMIT_COUNT]
+        log.debug("AcoustID: submitting batch of %d fingerprints...", len(submission_batch))
```

Comment:
I'd move those lines below and change debug message to show remaining fingerprints too.

```python
submission_batch = submissions[:self.BATCH_SUBMIT_COUNT]
submissions = submissions[self.BATCH_SUBMIT_COUNT:]
log.debug("AcoustID: submitting batch of %d fingerprints (%d remaining)...", len(submission_batch), len(submissions))
```